### ISSUE TITLE
Bug:cannot connect to camera

### ISSUE THREAD

**Issue by vdiogo** [JurajNyiri/HomeAssistant-Tapo-Control#42]
**Describe the bug**
I have installed the integration and added my camera credentials and two entities (SD and HD) were added in Hassio as expected. Issue is that I'm not able to connect to the cameras.
The camera works fine in the iPhone App

**To Reproduce**
1. Go to entity
2. Select SD or HD stream
3. Nothing happens (it keeps trying to connect)

**Expected behavior**
Would expect the camera to stream the video

**Log**
If applicable, add error logs.

```


Logger: custom_components.tapo_control
Source: helpers/update_coordinator.py:152
Integration: Tapo: Cameras Control (documentation, issues)
First occurred: 15:47:47 (1 occurrences)
Last logged: 15:47:47
Error requesting Tapo resource status data: HTTPSConnectionPool(host='192.168.86.70', port=443): Max retries exceeded with url: /stok=38c3b22570d75333a4e496c62ac0992b/ds (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x9f209688>: Failed to establish a new connection: [Errno 110] Operation timed out')) 
```

**Camera (please complete the following information):**
 - Device Model: C200
 - FW:1.0.16 Build 200930 Rel.26253n(4555)]

**Using stream component**
yes

**Camera works through onvif component OR different cameras work in general**
yes

**Camera has all attributes filled out properly in developer tools**
yes: user/ pass was created in the camera

**HASS Environment**
Docker

**Additional context**
N/A

**Comment by JurajNyiri** [comment]
Does your rtsp stream work? Try restarting the camera and make sure it is reachable on the ip specified.

Also open ports 443 and 2020 if using vlan.

**Comment by JurajNyiri** [comment]
You can also try using the home assistant rtsp or onvif component. This is 99.9% issue related to your network. IPhone app might work through cloud.

**Comment by vdiogo** [comment]
Where should I open the ports? I am assuming that because I am running the integration in the same network as the camera and home assistant, that all ports are open, or is that not the case?

What are the rtsp/ onvif components? Apologies, a bit noobie here

**Comment by JurajNyiri** [comment]
Yes that should be the case. Something must've changed something after adding the camera as when the camera is added, credentials are verified with the camera.

Maybe the camera ip changed? Make sure it has static ip.

You can also try to play rtsp stream through vlc for example. Instructions: https://www.tp-link.com/us/support/faq/2680/

Regarding the native integrations search for Home assistant onvif camera and home assistant rtsp camera on Google.

**Comment by JurajNyiri** [comment]
Closing because of issue not being related to this integration.

**Comment by vdiogo** [comment]
The camera started streaming. Didn't do anything so it's just for the record here.

**Comment by Ruco** [comment]
For anyone else finding this, with a similar problem.

I had to change my camera password to be able to get the stream to show up.
Password worked fine when using ONVIF Device Manager and VLC, but seems either this addon or the stream integration didn't like my previous password.

My previous password contained several non-alphanumeric characters(%!#), which is most likely what caused the issue.

**Comment by JurajNyiri** [comment]
@Ruco Bug with special characters in username/password is now be fixed in 2.3.1.
